core/node/node_sync/src/batch_transaction_updater/l1_transaction_verifier.rs

```diff
@@ -0,0 +1,390 @@
+                        self.diamond_proxy_addr
+                    );
+                    return None;
+                }
+                let parsed_log = event
+                                        .parse_log_whole(log.clone().into())
+
+                    .ok()?; // Skip logs that are of different event type
+
+                let block_number_from_log = get_param(&parsed_log.params, "batchNumber")
+                        .and_then(ethabi::Token::into_uint)
+                        .and_then(|x| u32::try_from(x).ok())
+                        .map(L1BatchNumber)
+                        .expect("Missing expected `batchNumber` parameter in `BlockCommit` event log");
+
+                if block_number_from_log != batch_number {
+                    tracing::warn!(
+                        "Commit transaction {0:?} has `BlockCommit` event log with batchNumber={block_number_from_log}, \
+                        but we are checking for batchNumber={batch_number}", receipt.transaction_hash
+                    );
+                    return None;
+                }
+
+                let batch_hash = get_param(&parsed_log.params, "batchHash")
+                        .and_then(ethabi::Token::into_fixed_bytes).map(|bytes| H256::from_slice(&bytes))
+                        .expect("Missing expected `batchHash` parameter in `BlockCommit` event log");
+
+                let commitment = get_param(&parsed_log.params, "commitment")
+                        .and_then(ethabi::Token::into_fixed_bytes).map(|bytes| H256::from_slice(&bytes))
+                        .expect("Missing expected `commitment` parameter in `BlockCommit` event log");
+
+                Some((batch_hash, commitment))
+            });
+
+        if let Some((batch_hash, commitment)) = commited_batch_info {
+            if db_batch.commitment != commitment {
+                return Err(TransactionValidationError::BatchTransactionInvalid { reason: format!("Commit transaction {} for batch {} has different commitment: expected {:?}, got {:?}",
+                    receipt.transaction_hash,
+                    batch_number,
+                    db_batch.commitment,
+                    commitment) });
+            }
+            if db_batch.root_hash != batch_hash {
+                return Err(TransactionValidationError::BatchTransactionInvalid { reason: format!("Commit transaction {} for batch {} has different root hash: expected {:?}, got {:?}",
+                    receipt.transaction_hash,
+                    batch_number,
+                    db_batch.root_hash,
+                    batch_hash)});
+            }
+            // OK verified successfully the commit transaction.
+            tracing::debug!(
+                "Commit transaction {} for batch {} verified successfully",
+                receipt.transaction_hash,
+                batch_number
+            );
+            Ok(())
+        } else {
+            Err(TransactionValidationError::BatchTransactionInvalid {
```

Comment:
Nit: You can transform `commited_batch_info` into a result and shortcut it using `ok_or_else`.